core/App/ehsm_provider.cpp

```diff
@@ -810,4 +811,44 @@ ehsm_status_t ExportDataKey(ehsm_keyblob_t *cmk,
         return EH_OK;
 }
 
+ehsm_status_t generate_apikey(uint8_t *p_apikey, uint32_t apikey_len, uint32_t *p_req_apikey_len)
+{
+    sgx_status_t sgxStatus = SGX_ERROR_UNEXPECTED;
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    if (p_req_apikey_len != NULL) {
+        // calculate length
+        ret = enclave_generate_apikey(g_enclave_id,
+                                    &sgxStatus,
```

Comment:
seem here, no need to call enclave_generate_apikey tiwce

Reply:
need get length from enclave.